SAP2000_Adapter/CRUD/Read/Bar.cs

```diff
@@ -119,9 +122,77 @@ private List<Bar> ReadBars(List<string> ids = null)
+                    double autoMeshMaxLength = 0.0;
+
+                    m_model.FrameObj.GetAutoMesh(id, ref autoMesh, ref autoMeshAtPoints, ref autoMeshAtLines, ref numSegs, ref autoMeshMaxLength);
+                    if (autoMesh)
+                    {
+                        bhomBar = bhomBar.SetAutoMesh(autoMesh, autoMeshAtPoints, autoMeshAtLines, numSegs, autoMeshMaxLength);
+                    }
+
+                    // Design Procedure
+
+                    int designProcedure = (int)DesignProcedureType.NoDesign;
+
+                    if (m_model.FrameObj.GetDesignProcedure(id, ref designProcedure) == 0)
+                    {
+                        DesignProcedureType designProcedureType = (DesignProcedureType)designProcedure;
+                        bhomBar = bhomBar.SetDesignProcedure(designProcedureType);
+                    }
+
+                    // Insertion Point Offset
+                    // Need to add more information to capture coordinate system?? GetCoordSys method and transform local csys
+                    int insertionPoint = (int)BarInsertionPoint.Centroid;
+                    bool mirror = false;
+                    bool modifyStiffness = false;
+
+                    double[] offset1 = new double[3];
+                    double[] offset2 = new double[3];
+                    string cSys = "";
+
+                    if (m_model.FrameObj.GetInsertionPoint(id, ref insertionPoint, ref mirror, ref modifyStiffness, ref offset1, ref offset2, ref cSys) == 0)
+                    {
+                        BarInsertionPoint barInsertionPoint = (BarInsertionPoint)insertionPoint;
+                        bhomBar = bhomBar.SetInsertionPoint(barInsertionPoint, modifyStiffness);
+                    }
+
+                    // Section Property Modifiers
+
+                    double[] sectionModifiers = new double[8];
+                    // first check if material assigned 
+                    if (m_model.FrameObj.GetModifiers(id, ref sectionModifiers) == 0)
+                    {
+                        SectionModifier sectionModifier = new SectionModifier();
+                        sectionModifier.Area = sectionModifiers[0];
+                        sectionModifier.Asz = sectionModifiers[1];
+                        sectionModifier.Asy = sectionModifiers[2];
+                        sectionModifier.J = sectionModifiers[3];
+                        sectionModifier.Iz = sectionModifiers[4];
+                        sectionModifier.Iy = sectionModifiers[5];
+                        // mass modifier = 6
+                        // weight modifier = 7
+                        if (bhomBar.SectionProperty == null)
+                        {
+                            ISectionProperty sectionProperty = new ExplicitSection();
+                            bhomBar.SectionProperty = sectionProperty;
+                        }
+                        bhomBar.SectionProperty.Fragments.Add(sectionModifier);
```

Comment:
SAP allows modifiers on the bar and on the section, vs. BHoM only allows them on the section. Section modifiers are already implemented in ReadSectionProperty(); this should not modify the section, but should create a modifier fragment on the bar with these values

Reply:
Correction: modifiers on sections are not implemented- only modifiers on SurfaceProperties. I believe section modifiers have been implemented in ETABS.

Reply:
Section modifiers are implemented in ETABS, but are on the section itself, not the bar.